test_cases/stdlib/asyncio/check_task.py

```diff
@@ -4,7 +4,7 @@
 
 
 class Waiter:
-    def __init__(self) -> None:
+    def __init__(self) -> None:  # pyright: ignore [reportMissingSuperCall] # Is this a false-positive?
```

Comment:
It is not a false positive. Every class can potentially be a subclass. For waiter,

class Bar:
  …
class Foo(Waiter, Bar):
  …

in Foo’s mro Waiter has parent class of Bar. This rule is very strict (key reason why it is not part of strict mode) as python multiple inheritance means you almost always need super call if class needs to worry about multiple inheritance usage.

Reply:
Oh I see! Thanks for the explanation. I realize I also had disabled it in my own projects a while ago but didn't understand why (I also wasn't quite familiar with python typing at the time).

So this rule just doesn't make sense in the context of type testing.